src/js/components/NasaTable.js

```diff
@@ -0,0 +1,117 @@
+            page: 0,
+            rowsPerPage: 10,
+            open: false,
+            onLink: false
+        }
+    }
+
+    handleClose = () => this.setState({ open: false })
+
+    getDataToShow = (table, from, rowsRepPage) => table.filter((row, i) => i >= from && i < from + rowsRepPage)
+
+    nextPage = async () => {
+        const { rowsPerPage, page, data } = this.state
+        const from = rowsPerPage * (page + 1)
+        const visibleRows = this.getDataToShow(data.table, from, rowsPerPage)
+        this.setState({ visibleRows, page: page + 1 })
+    }
+
+    prevPage = async () => {
+        const { rowsPerPage, page, data } = this.state
+        const from = rowsPerPage * (page - 1)
+        const visibleRows = this.getDataToShow(data.table, from, rowsPerPage)
+        this.setState({ visibleRows, page: page - 1 })
+    }
+
+    changeRow = async (e) => {
+        const { data } = this.state
+        const rows = +e.target.value
+        const visibleRows = this.getDataToShow(data.table, 0, rows)
+        this.setState({ rowsPerPage: rows, visibleRows, page: 0 })
+    }
+
+    fillCells = (cell, i) => {
+        if (i !== 0) {
+            return (<TableCell key={i}>{ cell }</TableCell>)
+        }
+        return (
+            <TableCell
+                key={i}
+                onClick={this.openDialog}
+                onMouseEnter={ () => this.setState({onLink: true})}
+                onMouseLeave={ () => this.setState({onLink: false})}
+                className={`first-item ${this.state.onLink && 'on-link'}`}
+            >
+                { cell }
+            </TableCell>)
+    }
+
+    openDialog = () => this.setState({ open: true })
+
+    componentDidMount = async () => {
+        const { rowsPerPage, page } = this.state;
+        const res = await getNasaData();
+        const data = await res.json();
+        const from = rowsPerPage * page
+        const visibleRows = this.getDataToShow(data.table, from, rowsPerPage)
+        this.setState({ data, visibleRows })
+    }
+
+    render() {
+        const { data, page, rowsPerPage, visibleRows, open } = this.state
+        const { headers, table = [] } = data
+        return(
+            <React.Fragment>
+                <ImageDialog open={ open } onClose={ this.handleClose }/>
+                <Table>
+                    <TableHead>
+                        <TableRow>
+                            { headers && headers.map((el, i) => <TableCell  className='table-headers' key={i}>{ el }</TableCell>)}
+                        </TableRow>
+                    </TableHead>
+                    <TableBody>
+                        { visibleRows && visibleRows.map((row, i) => <TableRow key={i}>
+                            { row.info.map((cell, i) => this.fillCells(cell, i)) }
+                        </TableRow>) }
+                    </TableBody>
+                </Table>
+                <TablePagination
+                    rowsPerPageOptions={[5, 10, 25]}
```

Comment:
move it to a separate class property please

Reply:
Can you clarify please your request, as far as I understood you want me to move [ 5, 10, 25 ] to constructor either you want to put it like props, now I will put it to constructor like this.rowsValue = [ 5, 10, 25 ].